wait/host.go

```diff
@@ -586,43 +586,60 @@ func (a *HostAwaitility) WaitForTemplateUpdateRequests(namespace string, count i
 // NotificationWaitCriterion checks if a Notification meets the given condition
 type NotificationWaitCriterion func(a *HostAwaitility, mur *toolchainv1alpha1.Notification) bool
 
-// WaitForNotification waits until there is a Notification available with the given name and the optional conditions
-func (a *HostAwaitility) WaitForNotification(name string, criteria ...NotificationWaitCriterion) (*toolchainv1alpha1.Notification, error) {
-	var notification *toolchainv1alpha1.Notification
+// WaitForNotifications waits until there is a Notification available with the given name and the optional conditions
+func (a *HostAwaitility) WaitForNotifications(name, reason string, criteria ...NotificationWaitCriterion) ([]*toolchainv1alpha1.Notification, error) {
+	var notifications []*toolchainv1alpha1.Notification
 	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
-		obj := &toolchainv1alpha1.Notification{}
-		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: name}, obj); err != nil {
+		labels := map[string]string{toolchainv1alpha1.NotificationUserNameLabelKey: name, toolchainv1alpha1.NotificationTypeLabelKey: reason}
+		opts := client.MatchingLabels(labels)
+		notificationList := &toolchainv1alpha1.NotificationList{}
+		if err :=  a.Client.List(context.TODO(), notificationList, opts);  err != nil {
 			if errors.IsNotFound(err) {
 				a.T.Logf("waiting for availability of notification '%s'", name)
 				return false, nil
 			}
 			return false, err
 		}
-		for _, match := range criteria {
-			if !match(a, obj) {
-				return false, nil
+
+		for _, n := range notificationList.Items {
+			for _, match := range criteria {
+				if match(a, &n) {
```

Comment:
Just a detail - do we really need to pass it as a pointer? This is the only usage of the `NotificationWaitCriterion` type right?

Reply:
Yeah, not too sure why it was a pointer, was just re-using what was already there. But i've changed it as part of this commit: 72d3c5c66de1c858dfee03ed84878e0fe86996de